### PR TITLE
Add jsonschema and protobuf support for ccompat api

### PR DESCRIPTION
**Pull request by carlesarnal**

### CODE REVIEW

File: app/src/main/java/io/apicurio/registry/ccompat/rest/impl/CompatibilityResourceImpl.java [hunk 1]
```diff
@@ -68,9 +68,8 @@ public CompatibilityCheckResponse testCompatibilityBySubjectName(
             SchemaContent request) throws Exception {
 
         CompatibilityCheckResponse result = facade.parseVersionString(subject, versionString, version -> {
-            // TODO are we safe to assume AVRO?
             try {
-                rules.applyRule(subject, version, ArtifactType.AVRO, ContentHandle.create(request.getSchema()));
+                rules.applyRule(subject, version, ArtifactType.fromValue(request.getSchemaType()), ContentHandle.create(request.getSchema()));
```
Comment (EricWittmann) [inline app/src/main/java/io/apicurio/registry/ccompat/rest/impl/CompatibilityResourceImpl.java:72]: Are Confluent's schema type constants the same as ours??

File: app/src/test/java/io/apicurio/registry/ConfluentClientTest.java [hunk 2]
```diff
@@ -16,38 +16,43 @@
...
 
-import java.nio.ByteBuffer;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.concurrent.ThreadLocalRandom;
-import java.util.function.BiConsumer;
-import java.util.function.Consumer;
+import java.util.*;
```
Comment (EricWittmann) [inline app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:46]: I would prefer to not sure wildcards for imports.

File: app/src/test/java/io/apicurio/registry/ConfluentClientTest.java [hunk 3]
```diff
@@ -184,81 +233,4 @@ public void testDelete() throws Exception {
...
-        converter.configure(config, false);
-
-        byte[] bytes = converter.fromConnectData(subject, cs, struct);
-
-        post.accept(client, bytes);
-
-        SchemaAndValue sav = converter.toConnectData(subject, bytes);
-        Struct ir = (Struct) sav.value();
-        Assertions.assertEquals("somebar", ir.get("bar").toString());
-    }
```
Comment (EricWittmann) [inline app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:263]: Why is this all removed?

Comment (EricWittmann) [reply app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:263]: And the converter classes too.  Why are they all gone?

File: app/src/test/java/io/apicurio/registry/ConfluentClientTest.java (hunk 2 above)
Comment (carlesarnal) [reply app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:46]: Automatically changed by the IDE, I'll change that.

File: app/src/main/java/io/apicurio/registry/ccompat/rest/impl/CompatibilityResourceImpl.java (hunk 1 above)
Comment (carlesarnal) [reply app/src/main/java/io/apicurio/registry/ccompat/rest/impl/CompatibilityResourceImpl.java:72]: Yep, same constant values.

File: app/src/test/java/io/apicurio/registry/ConfluentClientTest.java (hunk 3 above)
Comment (carlesarnal) [reply app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:263]: The converter has a comment from @alesj indicating that can be removed once confluent changes Avro version from `1.8.x` to `1.9.x` and that's what happened in the version `5.5.0.`

Comment (EricWittmann) [reply app/src/test/java/io/apicurio/registry/ConfluentClientTest.java:263]: OK great, thanks.

**Comment by EricWittmann** [comment]
I think we are missing a new Confluent endpoint:

https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-types-

Can you add that in as well?  Maybe also check for any other new endpoints that we might be missing.

**Comment by EricWittmann** [comment]
Also @carlesarnal it looks like CI is still failing - failing on the QE tests.  Can you try those tests locally and see what you get?  The errors seem to be unrelated to your change, but please verify.

**Comment by carlesarnal** [comment]
Sure, I'll have a look at both things (missing endpoint and QE tests).

**Comment by carlesarnal** [comment]
@EricWittmann, the last commit solves the problems with QE tests. The issue was related to [Jersey problem](https://github.com/Apicurio/apicurio-registry/issues/181). The obvious question here is, why is this test failing only in this PR and not in all the others?

**Comment by carlesarnal** [comment]
Hey @EricWittmann, I've noticed that we're also missing this [Version referenced by](https://docs.confluent.io/current/schema-registry/develop/api.html#get--subjects-(string-%20subject)-versions-versionId-%20version-referencedby) and this [Subject version pairs by id](https://docs.confluent.io/current/schema-registry/develop/api.html#get--schemas-ids-int-%20id-versions). Do you want me to add those endpoints as well?

**Comment by EricWittmann** [comment]
Yes please!

**Comment by carlesarnal** [comment]
Hi @EricWittmann, this is now ready for review. I've added the last two endpoints. Now we are in sync with the las version `5.5.0`.

**Pull request merged**